librz/util/utf8.c

```diff
@@ -1,204 +1,313 @@
-	{ 0x1EE28, 0x1EE28 }, { 0x1EE33, 0x1EE33 }, { 0x1EE38, 0x1EE38 },
-	{ 0x1EE3A, 0x1EE3A }, { 0x1EE3C, 0x1EE41 }, { 0x1EE43, 0x1EE46 },
-	{ 0x1EE48, 0x1EE48 }, { 0x1EE4A, 0x1EE4A }, { 0x1EE4C, 0x1EE4C },
-	{ 0x1EE50, 0x1EE50 }, { 0x1EE53, 0x1EE53 }, { 0x1EE55, 0x1EE56 },
-	{ 0x1EE58, 0x1EE58 }, { 0x1EE5A, 0x1EE5A }, { 0x1EE5C, 0x1EE5C },
-	{ 0x1EE5E, 0x1EE5E }, { 0x1EE60, 0x1EE60 }, { 0x1EE63, 0x1EE63 },
-	{ 0x1EE65, 0x1EE66 }, { 0x1EE6B, 0x1EE6B }, { 0x1EE73, 0x1EE73 },
-	{ 0x1EE78, 0x1EE78 }, { 0x1EE7D, 0x1EE7D }, { 0x1EE7F, 0x1EE7F },
-	{ 0x1EE8A, 0x1EE8A }, { 0x1EE9C, 0x1EEA0 }, { 0x1EEA4, 0x1EEA4 },
-	{ 0x1EEAA, 0x1EEAA }, { 0x1EEBC, 0x1EEEF }, { 0x1EEF2, 0x1EFFF },
-	{ 0x1F02C, 0x1F02F }, { 0x1F094, 0x1F09F }, { 0x1F0AF, 0x1F0B0 },
-	{ 0x1F0BF, 0x1F0C0 }, { 0x1F0D0, 0x1F0D0 }, { 0x1F0E0, 0x1F0FF },
-	{ 0x1F10B, 0x1F10F }, { 0x1F12F, 0x1F12F }, { 0x1F16C, 0x1F16F },
-	{ 0x1F19B, 0x1F1E5 }, { 0x1F203, 0x1F20F }, { 0x1F23B, 0x1F23F },
-	{ 0x1F249, 0x1F24F }, { 0x1F252, 0x1F2FF }, { 0x1F321, 0x1F32F },
-	{ 0x1F336, 0x1F336 }, { 0x1F37D, 0x1F37F }, { 0x1F394, 0x1F39F },
-	{ 0x1F3C5, 0x1F3C5 }, { 0x1F3CB, 0x1F3DF }, { 0x1F3F1, 0x1F3FF },
-	{ 0x1F43F, 0x1F43F }, { 0x1F441, 0x1F441 }, { 0x1F4F8, 0x1F4F8 },
-	{ 0x1F4FD, 0x1F4FF }, { 0x1F53E, 0x1F53F }, { 0x1F544, 0x1F54F },
-	{ 0x1F568, 0x1F5FA }, { 0x1F641, 0x1F644 }, { 0x1F650, 0x1F67F },
-	{ 0x1F6C6, 0x1F6FF }, { 0x1F774, 0x1FFFF }, { 0x2A6D7, 0x2A6FF },
-	{ 0x2B735, 0x2B73F }, { 0x2B81E, 0x2F7FF }, { 0x2FA1E, 0xF0000 },
-	{ 0xFFFFE, 0xFFFFF }, { 0x10FFFE, 0x10FFFF }, { 0x110000, 0xFFFFFFFF }
+	{ 0x0000, 0x001F }, // <control>
+	{ 0x007F, 0x009F }, // <control>
+	{ 0x00AD, 0x00AD }, // SOFT HYPHEN
+	{ 0x0600, 0x0605 }, // ARABIC NUMBER MARK ABOVE
+	{ 0x061C, 0x061C }, // ARABIC LETTER MARK
+	{ 0x06DD, 0x06DD }, // ARABIC END OF AYAH
+	{ 0x070F, 0x070F }, // SYRIAC ABBREVIATION MARK
+	{ 0x0890, 0x0891 }, // ARABIC POUND MARK ABOVE
+	{ 0x08E2, 0x08E2 }, // ARABIC DISPUTED END OF AYAH
+	{ 0x180E, 0x180E }, // MONGOLIAN VOWEL SEPARATOR
+	{ 0x200B, 0x200F }, // ZERO WIDTH SPACE
+	{ 0x202A, 0x202E }, // LEFT-TO-RIGHT EMBEDDING
+	{ 0x2060, 0x2064 }, // WORD JOINER
+	{ 0x2066, 0x206F }, // LEFT-TO-RIGHT ISOLATE
+	{ 0xD800, 0xDB7F }, // Non Private Use High Surrogate
+	{ 0xDB80, 0xDBFF }, // Private Use High Surrogate
+	{ 0xDC00, 0xDFFF }, // ow Surrogate
+	{ 0xE000, 0xF8FF }, // Private Use
+	{ 0xFEFF, 0xFEFF }, // ZERO WIDTH NO-BREAK SPACE
+	{ 0xFFF9, 0xFFFB }, // INTERLINEAR ANNOTATION ANCHOR
+	{ 0x110BD, 0x110BD }, // KAITHI NUMBER SIGN
+	{ 0x110CD, 0x110CD }, // KAITHI NUMBER SIGN ABOVE
+	{ 0x13430, 0x1343F }, // EGYPTIAN HIEROGLYPH VERTICAL JOINER
+	{ 0x1BCA0, 0x1BCA3 }, // SHORTHAND FORMAT LETTER OVERLAP
+	{ 0x1D173, 0x1D17A }, // MUSICAL SYMBOL BEGIN BEAM
+	{ 0xE0001, 0xE0001 }, // LANGUAGE TAG
+	{ 0xE0020, 0xE007F }, // TAGS
+	{ 0xF0000, 0xFFFFD }, // Plane 15 Private Use
+	{ 0x100000, 0x10FFFD }, // Plane 16 Private Use
+	{ 0x378, 0x379 }, { 0x380, 0x383 }, { 0x38b, 0x38b }, // Undefined
```

Comment:
Please sort intervals by start address in ascending order ... `rz_code_point_is_printable()` basically does a binary search.

Reply:
Oups, forgot this.